ta4j-examples/src/main/java/ta4jexamples/indicators/numeric/NumericIndicator.java

```diff
@@ -0,0 +1,232 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package ta4jexamples.indicators.numeric;
+
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.Indicator;
+import org.ta4j.core.Rule;
+import org.ta4j.core.indicators.EMAIndicator;
+import org.ta4j.core.indicators.SMAIndicator;
+import org.ta4j.core.indicators.helpers.ClosePriceIndicator;
+import org.ta4j.core.indicators.helpers.ConstantIndicator;
+import org.ta4j.core.indicators.helpers.HighestValueIndicator;
+import org.ta4j.core.indicators.helpers.LowestValueIndicator;
+import org.ta4j.core.indicators.helpers.PreviousValueIndicator;
+import org.ta4j.core.indicators.helpers.VolumeIndicator;
+import org.ta4j.core.indicators.statistics.StandardDeviationIndicator;
+import org.ta4j.core.num.Num;
+import org.ta4j.core.rules.CrossedDownIndicatorRule;
+import org.ta4j.core.rules.CrossedUpIndicatorRule;
+import org.ta4j.core.rules.OverIndicatorRule;
+import org.ta4j.core.rules.UnderIndicatorRule;
+
+/**
+ * NumericIndicator is a "fluent decorator" for Indicator<Num>. It provides
+ * methods to create rules and other "lightweight" indicators, using a
+ * (hopefully) natural-looking and expressive series of method calls.
+ * <p>
+ * Methods like plus(), minus() and sqrt() correspond directly to methods in the
+ * Num interface. These methods create "lightweight" (not cached) indicators to
+ * add, subtract, etc. Many methods are overloaded to accept either
+ * Indicator<Num> or Number arguments.
+ * <p>
+ * Methods like sma() and ema() simply create the corresponding indicator
+ * objects, (SMAIndicator or EMAIndicator, for example) with "this" as the first
+ * argument. These methods usually instantiate cached objects.
+ * <p>
+ * Another set of methods, like crossedOver() and isGreaterThan() create Rule
+ * objects. These are also overloaded to accept both Indicator<Num> and Number
+ * arguments.
+ */
+public class NumericIndicator implements Indicator<Num> {
```

Comment:
please rename it to `NumIndicator`.

Reply:
no...  :)   my name is nicer, if a little bit longer to type...  we should use user-facing (public API) names that reflect domain concepts or design intent, not implementation details...  I am old, my friend... I have struggled with choosing good names most of my career...  read it aloud....  a listener is going to stop you and ask what "num" means, even if they know what an "indicator" is

Reply:
@johnthetireless Even if you _feel_ it's nicer we should match and comply to this ta4j's naming scheme and ta4j is using `Num` It's irritating when we introduce again a new name for number/numeric. And looking at what you are implementing `implements Indicator<Num>`, you do nothing more than using the `Num` as an **indicator**. So would be nice to name the thing what it is: a `NumIndicator`.

Reply:
I am not going to change it.  If this PR is accepted you are certainly free to change it then.

Reply:
@johnthetireless I see that it resides within the  `numeric`-package and looking again I must confess, that `NumericIndicator` is indeed the right name for that. I also it better now:)